### PR TITLE
Set

### PR DESCRIPTION
**Pull request by shakalee14**

### CODE REVIEW

File: npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,49 @@
+0 info it worked if it ends with ok
```
Comment (rae-ralston) [inline npm-debug.log]: add the npm-debug.log to your .gitignore

File: spec/set.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import chaiAsPromised from 'chai-as-promised'
+import Set from '../src/set'
+
+chai.use(chaiChange)
+// chai.use(chaiAsPromised)
```
Comment (rae-ralston) [inline spec/set.js]: delete comments for PR

File: src/set.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    this._data.push(item)
+  }
+
+  isEmpty(){
+    return this._length === 0
+  }
+
+  contains(item){
+    for (let i = 0; i < this.size(); i++) {
+      if( this._data[i] === item ){
```
Comment (rae-ralston) [inline src/set.js:32]: ternary

**Comment by lumodon** [comment]
@Jusdev89 can you review this?
@shakalee14

File: src/set.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        }
+      }
+      if( (!alreadyInSet && union) || (alreadyInSet && !union)) {
+        resultSet._data.push( element )
+      }
+    })
+    return resultSet._data
+  }
+
+  union(otherSet){
```
Comment (rae-ralston) [inline src/set.js:80]: Nice, I like the way you did these parts :D

File: src/set.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    this.cloneTo(resultSet)
+
+    resultSet.forEach( (element) => {
+      for(let i = 0; i < otherSet.size(); i++) {
+        if(otherSet._data[i] === element) {
+          numMatching++
+        }
+      }
+    })
+    if(numMatching === resultSet.size()){
```
Comment (rae-ralston) [inline src/set.js:115]: return a ternary?

**Review by rae-ralston — COMMENTED** [review]
http://images5.fanpop.com/image/photos/30600000/Internet-High-Five-high-five-30662642-480-444.jpg

**Pull request merged**